### DISCUSSION TITLE
Capture activity life cycle on Appyx ?

### DISCUSSION THREAD

**Discussion by nicolas-f** [bumble-tech/appyx#687] [General]
Hi,

I have an appyx application but the application continue to do stuff event if it is in background:

https://github.com/nicolas-f/NoiseCaptureKotlin/blob/5db0be453c78e878890371dff2d36cfabc45b318/shared/src/commonMain/kotlin/org/noise_planet/noisecapture/shared/child/MeasurementScreen.kt#L205

How to capture such lifecycle events in my application. Or have a coroutine that run only on Activity Running state ?

![activity_lifecycle](https://github.com/bumble-tech/appyx/assets/1382241/8c36887e-acdd-4500-8691-07e7bf3fcd3d)

thank you

**Comment by zsoltk** [comment]
Hi @nicolas-f ,

I wonder if this would be helpful: https://bumble-tech.github.io/appyx/navigation/features/lifecycle/ 

When using a composable app hierarchy you don't need to focus on the Activity itself, but the current Node's own lifecycle (which is analogous in that the same concepts apply, just they're more granular).

**Reply by zsoltk** [reply]
Also see: https://bumble-tech.github.io/appyx/navigation/features/plugins/#lifecycle-related-plugins

**Reply by nicolas-f** [reply]
I tried with this:
```kotlin
lifecycle.asFlow().collect() {lifecycleState ->
                println("newState:${lifecycleState.name}")
            }
```
But there is no events related to stop or pause events:

```
newState:CREATED
// here i move the app in background (in android OnStop() and OnResume() has been called)
newState:STARTED
newState:RESUMED
```

**Reply by nicolas-f** [reply]
Ok the reason is that I need to implement the interface DefaultPlatformLifecycleObserver (with lifecycle.addObserver) in order to receive the fragment lifecycle. Thanks for the answer

```kotlin
interface DefaultPlatformLifecycleObserver : PlatformLifecycleObserver {
    fun onCreate() {}
    fun onStart() {}
    fun onResume() {}
    fun onPause() {}
    fun onStop() {}
    fun onDestroy() {}
}
```
